### PR TITLE
added keymaker documentation

### PR DESCRIPTION
**Pull request by snehalmaidneo**

### CODE REVIEW

File: modules/ROOT/pages/index.adoc [hunk 1]
```diff
@@ -92,6 +92,9 @@ The xref:aura-cli:index.adoc[Aura CLI] is a command line interface for interacti
 === xref:cypher-workbench:index.adoc[Cypher Workbench]
 The xref:cypher-workbench:index.adoc[Cypher Workbench] Cypher Workbench provides a set of web-based development graph tools. It includes tools for data modeling, cypher building and debugging, validation, and more. Originally a SaaS offering, the code is now released so you can run it in your local environment.
 
+=== xref:keymaker:index.adoc[Keymaker]
+The xref:keymaker:index.adoc[Keymaker] Keymaker is an innovative recommendations framework designed to accelerate the development of flexible Neo4j based applications.
```
Comment (jexp) [inline modules/ROOT/pages/index.adoc]: can you use a bit more descriptive text here? a bit more about what it does and what it consists of? this is too much marketing

File: modules/keymaker/pages/index.adoc [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+:author: Snehal Maid
+:category: labs
+:tags: graphql, api, cypher, webapp-development, cypher-query-pipelines
+:neo4j-versions: 4.4, 5.x
+:page-pagination:
+:page-product: keymaker
+
+image::keymaker_architecture.png[width=800]
+
+Keymaker is an innovative recommendations (next best action) framework designed to accelerate the development of flexible Neo4j based applications. It is a data model agnostic tool, It seamlessly integrates with Neo4j databases (versions 4.5.x and 5.x supported) allowing for the flexible configuration of analytical engines tailored to specific business needs across diverse industries. With Keymaker, you automatically gain access to a GraphQL API endpoint for each engine you develop, simplifying the process of constructing user interfaces with popular frameworks like React or System to System Interfaces.
```
Comment (jexp) [inline modules/keymaker/pages/index.adoc]: try to use a bit more technical language here, as if you explain it to a fellow colleague / developer

- what is it (technical)
- where / how would I use it in my stack
- what are the components
- how can I extend / modify / configure it

File: modules/keymaker/pages/index.adoc [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+== Installation
+
+* Check out the code from the https://github.com/neo4j-labs/keymaker[GitHub Repository]
+* Ensure that you have https://nodejs.org/[Node.js] on your machine
+* Follow the instructions in the Keymaker project README
+
+== Functionality Includes
+
+* *Seamless Integration with Neo4j Databases*. Supports integration with existing Neo4j databases, compatible with versions 4.5.x and 5.x.
+* *Score-Based Recommendation Engine*. Delivers personalized recommendations to users, leveraging scores generated based on predefined business logic.
```
Comment (jexp) [inline modules/keymaker/pages/index.adoc]: bit more detail?

File: modules/keymaker/pages/index.adoc [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+* *Simplified GraphQL Server Management*. Manages the GraphQL server infrastructure, enabling you to concentrate on defining business logic within engines, free from server management tasks.
+* *Dynamic Business Logic Modification*. Offers the capability for real-time adjustments to business logic, supporting immediate results review and A/B testing for optimized outcomes.
+* *Visualization Path Extraction for Neo4j’s Bloom*. Extracts and aggregates paths correlating with the results, facilitating their visualization through Neo4j’s Bloom product.
```
Comment (jexp) [inline modules/keymaker/pages/index.adoc]: can we add a screenshot of an app that uses keymaker as the backend and the graphql api type definition?

File: modules/keymaker/pages/index.adoc [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+
+== Documentation
+Documentation for this library is available link:https://help.neo4j.solutions/neo4j-solutions/keymaker/[here].
```
Comment (jexp) [inline modules/keymaker/pages/index.adoc]: use ^ at the end of the URL text to open the link in a new window (see below)

File: modules/keymaker/pages/index.adoc [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+<iframe width="320" height="180" src="https://www.youtube.com/embed/gUEHIiFAMrM" frameborder="0" allow="accelerometer; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
+++++
+
+Video quick start on keymaker:
+++++
+<iframe width="320" height="180" src="https://www.youtube.com/embed/NLaL0BN3SOs" frameborder="0" allow="accelerometer; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
+++++
+
+== Highlighted Articles
+link:https://medium.com/@snehal.maid/keymaker-dc496d41b21f/[Keymaker]
```
Comment (jexp) [inline modules/keymaker/pages/index.adoc]: Better title for the blog post

**Review by jexp — CHANGES_REQUESTED** [review]
see my comments

File: modules/ROOT/pages/index.adoc (hunk 1 above)
Comment (snehalmaidneo) [reply modules/ROOT/pages/index.adoc]: New text : Keymaker includes an interface which allows users to build analytical pipelines ( or engines ) using Cypher and automatically makes the results of the analytical pipeline available through a GraphQL API endpoint.

File: modules/keymaker/pages/index.adoc (hunk 2 above)
Comment (snehalmaidneo) [reply modules/keymaker/pages/index.adoc]: Added a high level architecture diagram for covering the various components, added a user activity flow diagram and edited the description to make it more technical and concise

File: modules/keymaker/pages/index.adoc (hunk 3 above)
Comment (snehalmaidneo) [reply modules/keymaker/pages/index.adoc]: Modified the blog to deep dive into the benefits of keymaker. The blog has a sample query pipeline and explains the response of advantages of keymaker.

File: modules/keymaker/pages/index.adoc (hunk 4 above)
Comment (snehalmaidneo) [reply modules/keymaker/pages/index.adoc]: Added grapgql api definition and screenshot in the blog

File: modules/keymaker/pages/index.adoc (hunk 5 above)
Comment (snehalmaidneo) [reply modules/keymaker/pages/index.adoc]: Done for all the urls

File: modules/keymaker/pages/index.adoc (hunk 6 above)
Comment (snehalmaidneo) [reply modules/keymaker/pages/index.adoc]: Modified the titles and added a subtitle to the blog

**Review by jexp — APPROVED** [review]
Loosk great, thank you

**Pull request merged**